src/main/java/homework9/ArrayValueCalculator.java

```diff
@@ -0,0 +1,29 @@
+package homework9;
+
+public class ArrayValueCalculator {
+
+    public int doCalc(String[][] inputArray) {
+        if (inputArray.length != 4 ||
```

Comment:
The validation statements are correct, but readability suffers :) 
Would be good to extract it to some method, that returns `boolean` and give it a name like `isMatrixSizeCorrect()`. The benefit of this approach is in a correct name of a method that checks your input. So, by reading a well-named method there might be no reason for another developer or whoever is gonna use this code to dive deeper into the code to check how `isMatrixSizeCorrect()` was implemented.